api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -495,17 +495,24 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
 
   @Override
   public ResponseEntity<WorkspaceResponseListResponse> getWorkspaces() {
-    // TODO: use FireCloud to determine what workspaces to return, instead of just returning
-    // workspaces from our database.
+    org.pmiops.workbench.db.model.Workspace dbWorkspace;
+    List<org.pmiops.workbench.firecloud.model.Workspace> fcWorkspaces;
     User user = userProvider.get();
     List<WorkspaceResponse> responseList = new ArrayList<WorkspaceResponse>();
     if (user != null) {
-      for (WorkspaceUserRole userRole : user.getWorkspaceUserRoles()) {
-        // TODO: Use FireCloud to determine access roles, not our DB
+      try {
+        fcWorkspaces = firecloudService.getWorkspaces();
```

Comment:
This is going to return all workspaces, not just AofU ones... including public workspaces.

I thought we were potentially going to defer this work until FC provided a way of filtering this endpoint by tags?

Reply:
We were, but its looking like we should apply a bandaid of some sort for now. We also discussed that even when they do add that, we are going to need to validate against our own DB, because nothing stops a user from going into FC and making a workspace there with our tag. (And then we don't capture research purpose/other metadata.) So if we need validation logic anyway, we may as well get on it now...

Reply:
So I see 110 public workspaces when I sign into Firecloud. What's the latency for fetching those? Are there going to be more public workspaces in future?

If it performs OK now I guess it's fine to make this change now, but the only difference to researchers is things are going to be slower. :(

Reply:
(Basically, I think this change -- making FC the source of truth about workspaces -- is a requirement for launch, but not especially pressing.)